holidays/countries/thailand.py

```diff
@@ -558,8 +558,9 @@ def _populate_government_holidays(self):
             2021: (MAY, 13),
             2022: (MAY, 17),
             2023: (MAY, 11),
+            2024: (MAY, 10),
         }
-        if 1957 <= self._year <= 2023 and self._year != 1999:
+        if 1957 <= self._year <= 2024 and self._year != 1999:
```

Comment:
Could you explain the year bump? Is it something that can be removed?

Reply:
The current logic either assigns the "Royal Ploughing Ceremony" date from the `raeknakhwan_dates` or `(MAY, 13)` for no-source-given years as rough approximates; this is mainly done so that May 13 isn't automatically assigned for future dates - at least until we've cracked the logic [the Court Astrologers/Royal Brahmins](https://www.bangkokpost.com/life/social-and-lifestyle/726736/the-new-brahmins) used for their calculations.

```python
        raeknakhwan_dates = {
            1997: (MAY, 13),
                  .
                  .
            2024: (MAY, 10),
        }
        if 1957 <= self._year <= 1996 and self._year != 1999:
            self._add_observed(
                # Royal Ploughing Ceremony.
                self._add_holiday(tr("วันพืชมงคล"), raeknakhwan_dates.get(self._year, (MAY, 13)))
            )
```

Reply:
Let me put it this way: do we want continue manual bumps every year or can remove the upper bound to allow future estimates?

Reply:
@arkid15r Ok, I did end up checking out all exact date instances of this from 1960-2024 - still can't find an exact pattern though some Wikipedia editor claimed that 2025 date for this would be [May 9](https://th.wikipedia.org/wiki/9_%E0%B8%9E%E0%B8%A4%E0%B8%A9%E0%B8%A0%E0%B8%B2%E0%B8%84%E0%B8%A1) with no source attached.

I've found a Thai phone number (02-6285218) listed in older Royal Gazette releases for this exact question. I did end up calling them, but that number is no longer available:
![image](https://github.com/vacanza/python-holidays/assets/19505219/26f43e21-4549-4831-a95e-1937acb9cc58)

I still wouldn't want to input my own rough approximation for this lib, especially when even the government aren't sure of it either besides "This roughly falls on May, always manually appointed by court astrologers and released in the Royal Calendar (ปฏิทินหลวง)".
![image](https://github.com/vacanza/python-holidays/assets/19505219/bb00a443-e127-431a-8798-b0f35b45f7e9)

There's no good online copy of the Royal Calendar (ปฏิทินหลวง) either, as they're given in physical copies-only for those who went to offer best wishes at certain Royal Residences (ลงนามถวายพระพร) in person on the New Year's Day.

Reply:
Oh wow, what a mess!
I'm sorry you had to do that, I didn't mean to make you jump through hoops. The idea was to minimize future edits.

I guess we could add (estimated) label if you change your mind about adding just estimates.

Reply:
Here's the actual statistics I got from running python code for 1960-2024 entry in case we want to try figure this out later:
> Monday: 14 times
Tuesday: 0 times
Wednesday: 13 times
Thursday: 17 times
Friday: 21 times
Saturday: 0 times
Sunday: 0 times

>Day 02: 1 times
Day 07: 7 times
Day 08: 7 times
Day 09: 10 times
Day 10: 9 times
Day 11: 12 times
Day 12: 2 times
Day 13: 7 times
Day 14: 4 times
Day 15: 1 times
Day 16: 2 times
Day 17: 2 times
Day 19: 1 times
![image](https://github.com/vacanza/python-holidays/assets/19505219/be3d0b55-97ca-48cf-8176-17fdc796bfff)

And here's them in terms of classical Thai calendar format, still very random IMO
![image](https://github.com/vacanza/python-holidays/assets/19505219/a6c84e23-2bb5-4dce-9bf2-8427cc71d0e6)

Reply:
Update 2: found a source here that really narrows down our date assignment that's actually usable for estimation, still not that suitable for python-ification though as that would requires PyMeeus. [Thankfully someone already did code a web version up to 2157 C.E.](https://www.myhora.com/%e0%b8%9b%e0%b8%8f%e0%b8%b4%e0%b8%97%e0%b8%b4%e0%b8%99/%e0%b8%9b%e0%b8%8f%e0%b8%b4%e0%b8%97%e0%b8%b4%e0%b8%99%e0%b9%82%e0%b8%ab%e0%b8%a3%e0%b8%b2%e0%b8%a8%e0%b8%b2%e0%b8%aa%e0%b8%95%e0%b8%a3%e0%b9%8c%e0%b9%84%e0%b8%97%e0%b8%a2-%e0%b8%9e%e0%b8%a4%e0%b8%a9%e0%b8%a0%e0%b8%b2%e0%b8%84%e0%b8%a1-%e0%b8%9e.%e0%b8%a8.2565.aspx) so manually assigning the likely future dates could work for now.

> ฤกษ์การพระราชพิธีนี้ ต้องหาฤกษ์วิเศษกว่าฤกษ์อื่นๆ คือ
> (๑) ฤกษ์นั้นอย่าให้ต้องวันผีเพลียอย่างหนึ่ง ดิถีซึ่งนับผีเพลียนั้นข้างขึ้นคือ ๑ ๕ ๗ ๘ ๙ ๑๐ ๑๑ ๑๕ ข้างแรม ๑ ๕ ๖ ๗ ๘ ๑๓ ๑๔ เป็นใช้ไม่ได้
(๒) ให้ได้ศุภดิถีอย่างหนึ่ง คือ ดิถีตาว่างซึ่งไม่เป็นผีเพลียนั่นเอง
(๓) ให้บุรณฤกษ์อย่างหนึ่ง คือ ๒ ๔ ๕ ๖ ๘ ๑๑ ๑๔ ๑๗ ๒๒ ๒๔ ๒๖ ๒๗
(๔) ให้ได้วันสมภเคราะห์อย่างหนึ่ง คือ วันจันทร์ วันพุธ วันพฤหัสบดี วันศุกร์

> Due to its significance of heralding new planting season, this are usually assigned to the date with most significant luck to it as seen in Point 3. (in the 6th Thai Classical Month: May in Gregorian calendar)
> 1. The following dates are "forbidden": 
     - Waxing Phase: 1, 5, 7, 8, 9, 10, 11, 15th (usable: 2-3-4-6-12-13-14th)
     - Waning Phase: 1, 5, 6, 7, 8, 13, 14th (usable: 2-3-4-9-10-11-12-15th)
> 2. The lunar phase must at least be in the "ดิถีตาว่าง" period. (see Point 1 usable list)
> 3. The final luck value as seen from all planetary positions (บุรณฤกษ์: Sun to Saturn and 3 more astrological derivative calculations) must either be 2, 4, 5, 6, 8, 11, 14, 17, 22, 24, 26 or 27
> 4. The date can only be assigned to Monday, Wednesday, Thursday, or Friday.

![image](https://github.com/vacanza/python-holidays/assets/19505219/0621d9a4-77c1-49e6-8a01-f5689d013310)

Of which I did a quick check for 2018-2024, only exception so far being 2023 but that's good enough for our usage here than simply assuming May, 13 for all of them.